cookiecutter_project_upgrader/cli.py

```diff
@@ -45,3 +54,29 @@ def _load_context(context_file: str):
         exit(1)
     context = json.loads(context_str)
     return context
+
+
+def _is_valid_file(zip_file: str):
+    # we check only for local files for valid, for remote URLs, cookiecutter can do the test if its valid.
+    REPO_REGEX = re.compile(
+        r"""
+        ((((git|hg)\+)?(git|ssh|file|https?):(//)?)
+        |
+        (\w+@[\w\.]+)
+        )
+        """,
+        re.VERBOSE,
+        )
+    try:
+        if not bool(REPO_REGEX.match(zip_file)):
+            if ZipFile(Path(zip_file)).testzip() is None:
+                return True
```

Comment:
```suggestion
    zip_path = Path(zip_file)
    try:
        if zip_path.exists():
            ZipFile(zip_path).testzip()
```

I believe this will have the same behaviour (always True, unless `testzip` throws an error), and avoids having a regexp to maintain :)

I don't think any of the formats from the regexp would actually fail to parse into a Path: 
```pycon
>>> Path('git+https://github.com').exists()
False
```

Reply:
Yes sure, using above we don't need to except FileNotFound error.